envs/linc/jupyterhub-overrides.yaml

```diff
@@ -0,0 +1,100 @@
+                  for val in auth_model["auth_state"]["scope"]:
+                      scopes.extend(val.split(","))
+                  auth_model["auth_state"]["scope"] = scopes
+              auth_model = await self.update_auth_model(auth_model)
+
+              # Allowed if admin
+              if await super().check_allowed(username, auth_model):
+                  return True
+
+              # Allowed if user is a registered DANDI user.
+              req = HTTPRequest(
+                  f"${dandi_api_domain}/api/users/search/?username={username}",  # noqa
+                  method="GET",
+                  headers={
+                      "Accept": "application/json",
+                      "User-Agent": "JupyterHub",
+                      "Authorization": "token ${danditoken}",
+                  },
+                  validate_cert=self.validate_server_cert,
+              )
+              try:
+                  client = AsyncHTTPClient()
+                  print(f"Attempting to validate {username} with ${dandi_api_domain}")  # noqa
+                  resp = await client.fetch(req)
+              except HTTPClientError as e:
+                  print(
+                      f"Dandi API request to validate {username} returned HTTPClientError: {e}"
+                  )
+                  return False
+              else:
+                  if resp.body:
+                      resp_json = json.loads(resp.body.decode("utf8", "replace"))
+                      for val in resp_json:
+                          if val["username"].lower() == username.lower():
+                              return True
+
+              # If not explicitly allowed, not allowed.
+              return False
+
+          async def pre_spawn_start(self, user, spawner):
+              auth_state = await user.get_auth_state()
+              if not auth_state:
+                  # user has no auth state
+                  return
+              # define some environment variables from auth_state
+              spawner.environment["GITHUB_TOKEN"] = auth_state["access_token"]
+              spawner.environment["GITHUB_USER"] = auth_state["github_user"]["login"]
+              spawner.environment["GITHUB_EMAIL"] = auth_state["github_user"]["email"]
+      c.KubeSpawner.modify_pod_hook = modify_pod_hook  # noqa
+      c.JupyterHub.authenticator_class = IsDandiUserAuthenticator  # noqa
+      c.GitHubOAuthenticator.enable_auth_state = True  # noqa
+
+singleuser:
+  lifecycleHooks:
+    postStart:
+      exec:
+        command:
+          - "sh"
+          - "-c"
+          - >
+            gitpuller https://github.com/dandi/example-notebooks master dandi-notebooks;
+            chown -R jovyan:users dandi-notebooks;
+            datalad install https://github.com/dandi/dandisets;
+            /opt/conda/envs/allen/bin/python -m ipykernel install --user --name allen --display-name "Python (Allen SDK)";
+            /opt/conda/bin/pip install --upgrade dandi;
```

Comment:
```suggestion
            /opt/conda/bin/pip install --upgrade dandi;
            /opt/conda/bin/pip install --upgrade lincbrain-cli;
```

Reply:
I deleted my previous comment and am keeping the upgrade command for the `dandi` package.